app/models/keyword.rb

```diff
@@ -11,4 +11,25 @@ class Keyword < ApplicationRecord
             presence: true,
             numericality: { greater_than_or_equal_to: 0 },
             if: :succeeded?
+
+  after_create_commit :enqueue_scrape_job
+  after_update_commit :broadcast_status_update
+
+  private
+
+  def enqueue_scrape_job
+    ScrapeKeywordJob.perform_later(keyword_id: id)
+  end
+
+  def broadcast_status_update
```

Comment:
While Rails makes it powerful to use model hooks to trigger actions, it is a tool we should use sparingly as 1) it is hard to test and 2) it mixes too much business logic (this use cases) with business entities. Moving this kind of use case to the form or service classes is better than placing them in the models. It si nothing wrong per se but it is a better application architecture choice.

To recap, using `after_create_commit` is fair use, but it expected business logic for all keywords.  Using `after_update_commit` to broadcast an update to the UI would be better in the service class `Google::ScrapeService`, especially as it is odd for a model to call a presenter class 🤔 (only decorators or job classes are usually called in models).

Reply:
Fixed in d1849b45b94658f7ff80415f783db57423b27cb4

Good point. I refactored the status broadcasting logic in to the ScraperService. 

But I'm wondering about the situation where another service class needs to broadcast keyword updates. Should it duplicate this logic in their own class. What do you think about extracting it to a `KeywordBroadcastable` concern? Obviously this is overkill for the current situation.

Reply:
Concerns as like mixins/modules so yes that would be the way to go not to duplicated code 👍